src/Service.Tests/CosmosTests/MutationTests.cs

```diff
@@ -321,11 +340,87 @@ public async Task UpdateItemWithUnauthorizedWildCardReturnsError()
                 name = "new_name"
             };
 
-            JsonElement response = await ExecuteGraphQLRequestAsync("updateEarth", mutation, variables: new() { { "id", id }, { "partitionKeyValue", id }, { "item", update } });
+            string authtoken = AuthTestHelper.CreateStaticWebAppsEasyAuthToken(specificRole: roleName);
+            JsonElement response = await ExecuteGraphQLRequestAsync(
+                queryName: "updateEarth",
+                query: mutation,
+                variables: new() { { "id", id }, { "partitionKeyValue", id }, { "item", update } },
+                authToken: authtoken,
+                clientRoleHeader: roleName);
 
-            // Validate the result contains the GraphQL authorization error code.
-            string errorMessage = response.ToString();
-            Assert.IsTrue(errorMessage.Contains(DataApiBuilderException.GRAPHQL_MUTATION_FIELD_AUTHZ_FAILURE));
+            Console.WriteLine(response.ToString());
+            if (string.IsNullOrEmpty(expectedErrorMessage))
+            {
+                Assert.AreEqual(id, response.GetProperty("id").GetString());
+            }
+            else
+            {
+                // Validate the result contains the GraphQL authorization error code.
+                string errorMessage = response.ToString();
+                Assert.IsTrue(errorMessage.Contains(expectedErrorMessage));
+            }
+        }
+
+        /// <summary>
+        /// Delete Mutation performed on the fields with different auth permissions
+        /// It throws permission denied error if role doesn't have permission to perform the operation
+        /// </summary>
+        [TestMethod]
+        [DataRow("field-mutation-with-read-permission", MutationTests.NO_ERROR_MESSAGE, DisplayName = "AuthZ success and blank response for delete mutation because of reference to excluded/disallowed fields.")]
+        [DataRow("authenticated", MutationTests.NO_ERROR_MESSAGE, DisplayName = "AuthZ success and blank response when role has no delete operation restrictions.")]
```

Comment:
isn't clear how these two tests are different. Even though the comments allude to the role restrictions being different, the response is blank for both tests. Is the response being blank for both tests intended?

Reply:
@sajeetharan is already reviewing these tests. These tests are showing current state of the authorization feature in cosmos db. If anything needs to be changed, I will change it is separate PR.

Reply:
That's fine he is reviewing the tests. I'm not suggesting a change needs to be made here though would like to know how the tests are different. 

I don't understand what restrictions are present just from reading this:
>   [DataRow("field-mutation-with-read-permission", MutationTests.NO_ERROR_MESSAGE, DisplayName = "AuthZ success and blank response for delete mutation because of reference to excluded/disallowed fields.")]

are the "because of reference to excluded/disallowed fields." on the read action or the delete action?

Reply:
Yes..if `exclude`and `include` has any effect on operation